### PR TITLE
Interface stream fixes

### PR DESCRIPTION
**Pull request by ivelten**
With this PR, I am proposing a solution for a bug when handling deferred fields inside interfaces.

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -8,6 +8,10 @@ nuget Newtonsoft.Json
 nuget System.Runtime
 nuget FSharp.Core
 
+group Samples
```
Comment (ivelten) [inline paket.dependencies:11]: Added this becase relay starter kit needs Suave, which was not referenced.

File: src/FSharp.Data.GraphQL.Server/Schema.fs [hunk 2]
```diff
@@ -106,8 +106,8 @@ type Schema<'Root> (query: ObjectDef<'Root>, ?mutation: ObjectDef<'Root>, ?subsc
         | Some c -> c
 
     let typeMap : TypeMap =
-        let m = mutation |> function Some(Named n) -> [n] | _ -> []
-        let s = subscription |> function Some(Named n) -> [n] | _ -> []
+        let m = mutation |> function Some (Named n) -> [ n ] | _ -> [ ]
```
Comment (johnberzy-bazinga) [inline src/FSharp.Data.GraphQL.Server/Schema.fs]: Can we remove the spaces between the `[ ]`

File: src/FSharp.Data.GraphQL.Server/Schema.fs [hunk 3]
```diff
@@ -106,8 +106,8 @@ type Schema<'Root> (query: ObjectDef<'Root>, ?mutation: ObjectDef<'Root>, ?subsc
...
+        let s = subscription |> function Some (Named n) -> [ n ] | _ -> [ ]
```
Comment (johnberzy-bazinga) [inline src/FSharp.Data.GraphQL.Server/Schema.fs]: Same as above

File: src/FSharp.Data.GraphQL.Server/Execution.fs [hunk 4]
```diff
@@ -425,22 +425,28 @@ let rec private buildResolverTree (returnDef: OutputDef) (ctx: ResolveFieldConte
...
     let rec build (acc: ResolverTree list) = function
         | info::xs ->
             let argDefs = fieldExecuteMap.GetArgs(objdef.Name, info.Definition.Name)
             let fieldCtx = createFieldContext objdef argDefs ctx info
             let execute = fieldExecuteMap.GetExecute(objdef.Name, info.Definition.Name)
-            resolveField execute fieldCtx value
-            |> AsyncVal.bind(buildResolverTree info.ReturnDef fieldCtx fieldExecuteMap)
-            |> AsyncVal.rescue(fun e -> ResolverError{ Name = info.Identifier; Message = ctx.Schema.ParseError e; PathToOrigin = []})
-            |> AsyncVal.bind(fun tree ->
+            let x = resolveField execute fieldCtx value
```
Comment (johnberzy-bazinga) [inline src/FSharp.Data.GraphQL.Server/Execution.fs]: We should either have something more descriptive than `x` here or pipe it to the next line.

**Pull request merged**